src/internal/tasks/task.ts

```diff
@@ -52,7 +52,7 @@ export abstract class Task {
     this._invocationEvent = invocationEvent;
 
     if (this.isDone()) {
-      this.finishExecution();
+      this._executionLock.release();
```

Comment:
Why don't you delete the task from the queue? 🤔

Reply:
Unless I'm missing something, it should be no longer needed, since the only way that condition can be fulfilled is if by that time the task has been cancelled, and during the cancelation the queue is cleared. Right?

Reply:
Yes, you are right! Thanks for the clarification 🙂

Reply:
You're welcome!! Thanks to you for carefully reviewing this :)